lib/Doctrine/DBAL/Schema/Comparator.php

```diff
@@ -432,12 +432,10 @@ public function diffColumn(Column $column1, Column $column2)
             $changedProperties[] = 'comment';
         }
 
-        if ($properties1['default'] !== $properties2['default'] ||
-            // Null values need to be checked additionally as they tell whether to create or drop a default value.
-            // null != 0, null != false, null != '' etc. This affects platform's table alteration SQL generation.
-            ($properties1['default'] === null && $properties2['default'] !== null) ||
-            ($properties2['default'] === null && $properties1['default'] !== null)
-        ) {
+        // Null values need to be checked additionally as they tell whether to create or drop a default value.
+        // null != 0, null != false, null != '' etc. This affects platform's table alteration SQL generation.
+        if (($properties1['default'] === null) !== ($properties2['default'] === null)
```

Comment:
@morozov not sure that's not my problem but with annotation like this:
```
@ORM\Column(type="boolean", nullable=false, options={"default" = false})
```
it always generates migration becouse of this fix. compared with old version https://3v4l.org/HBLG7 :
````php
<?php
$changedProperties = [];
$properties1['default'] = false;
$properties2['default'] = 0;

if ($properties1['default'] != $properties2['default'] ||
    (null === $properties1['default'] && null !== $properties2['default']) ||
    (null === $properties2['default'] && null !== $properties1['default'])
) {
    $changedProperties[] = 'default';
}

var_dump($changedProperties);

if (($properties1['default'] === null) !== ($properties2['default'] === null)
    || (string) $properties1['default'] !== (string) $properties2['default']) {
    $changedProperties[] = 'default';
}

var_dump($changedProperties);
````
result:
```
array(0) {
}
array(1) {
  [0]=>
  string(7) "default"
}
```

Reply:
If we do not support `true/false` values for boolean type maybe do not allow them?

Reply:
Because `('0' == false) === true` but `('0' === (string) false) === false`.

![](https://user-images.githubusercontent.com/144181/49427589-f65fbf00-f7a3-11e8-98e2-8d906a10a8fa.png)

Reply:
@Majkl578 yes yes... but should I update my default value to `0` or this will be fixed?

Reply:
I think we'll loosen `===` back to `==` for the patch release, but later, the missing component here seems to be `Type::convertToDatabaseValue()`. We're loosely comparing a PHP value with a DB value which may have its own not yet discovered edge cases.